assets/src/i18n/translateWithMarkup.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { cloneElement, createElement } from 'react';
+import PropTypes from 'prop-types';
+
+/** @typedef {import('react').React.ReactNode} ReactNode */
+/** @typedef {import('react').React.ReactElement} ReactElement */
+
+// See https://html.spec.whatwg.org/multipage/syntax.html#void-elements
+const VOID_ELEMENTS = [
+  'area',
+  'base',
+  'br',
+  'col',
+  'embed',
+  'hr',
+  'img',
+  'input',
+  'link',
+  'meta',
+  'param',
+  'source',
+  'track',
+  'wbr',
+];
+
+/**
+ * Transforms a single DOM node.
+ *
+ * @param {Node} node DOM node.
+ * @param {?Object.<string, ReactElement>} mapping Map of tag names to React components.
+ * @return {ReactNode} Transformed node.
+ */
+export function transformNode(
+  { childNodes, localName, nodeType, textContent },
+  mapping = {}
+) {
+  if (Node.TEXT_NODE === nodeType) {
+    return textContent;
+  }
+
+  const children = childNodes?.length
+    ? [...childNodes].map((child) => transform(child, mapping))
+    : textContent || null;
```

Comment:
Can it have `textContent` if not a `TEXT_NODE`? Is it a comment then?

Reply:
A regular node with only text in it but no child elements will have textContent but no children

Reply:
Actually, scratch that. Seems to be some leftover from testing.